### DISCUSSION TITLE
I would like to be able to provide a list of URLs, and/or upload documentation manuals to allow Cline to look up APIs, etc...

### DISCUSSION THREAD

**Discussion by phalexo** [cline/cline#1088] [Feature Requests]
For example, if I am developing a web app, and using node, Express, Playwright, Stripe payments, I want to provide all the right versions of the software packages' documentation. Language models are often confused about which specific release is being used and make a total mess of it.

So, some kind of documentation RAG for Cline would be good.

**Comment by AviOfLagos** [comment]
exactlyyy!!!
Like a local knowledge base


I tried to build an MCP tool for that with cline, and a way it can help reduce hallucination and suggest fixes... but i dont think it worked :-(

**Comment by phalexo** [comment]
I made a little progress with Express and Playwright (the tools return
something when I test them), total failure on Stripe.
It seems that some community MCP implementations include webcrawlers. Maybe
it is possible to build on top of those,
but it is not even clear to me how to use the community MCP servers.

**Comment by andrewhopper** [comment]
Building Hivemind to solve for this - https://github.com/andrewhopper/hivemind

**Comment by sdi2200262** [comment]
Thanks for the detailed feedback. Most of what you're describing is addressed in v1.0.0, which is a ground-up redesign of the workflow.

For customizing APM's behavior (init preferences, subagent permissions, execution patterns), v1 supports **custom repositories**. You fork the repo, modify the templates to fit your needs, and install with `apm custom -r owner/repo`. A built-in customization skill guides AI agents through the process.

See the [Customization Guide](https://agentic-project-management.dev/docs/customization-guide) for details and the [full v1 docs](https://agentic-project-management.dev) for the redesigned workflow.
